### PR TITLE
fix(ui): Refresh issue tags when environment selection is changed

### PR DESCRIPTION
**Pull request by iProgramStuff**
On an issue details page while viewing the tags, switching environments does not update the tag count. This PR changes it so that it does.

Fixes SEN-979 & ISSUE-535 & SEN-980 & ISSUE-567

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx [hunk 1]
```diff
@@ -35,14 +37,20 @@ class GroupTags extends React.Component {
...
+  componentDidUpdate(prevProps) {
+    if (!isEqual(prevProps.environments, this.props.environments)) {
+      this.fetchData();
+    }
+  }
+
   fetchData = () => {
-    const {api, group} = this.props;
+    const {api, group, environments} = this.props;
+    const env_query = environments.map(e => '&environment=' + e).join('');
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx]: nit: use `camelCase`

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx]: I think more correct would be 
```
environments.map(e => 'environment=' + e).join('&')
```
So you don't have the leading `&`

But I think even better would be to not stringify URL parameters at all and pass them in the 2nd parameter of `api.request` using the `data` key.

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx [hunk 2]
```diff
@@ -35,14 +37,20 @@ class GroupTags extends React.Component {
...
     this.setState({
       loading: true,
       error: false,
     });
-
-    api.request('/issues/' + group.id + '/tags/', {
+    api.request('/issues/' + group.id + '/tags/?' + env_query, {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx]: This is old code, but can we use template strings here? 

```suggestion
    api.request(`/issues/${group.id}/tags/`, {
```

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTags.jsx]: Thanks Billy. Will note the camelCase preference. Removed the stringify'd parameters in favour of using the query key.

File: tests/js/spec/views/organizationGroupDetails/groupTags.spec.jsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ describe('GroupTags', function() {
       <GroupTags
         group={group}
         query={{}}
+        environments={['dev']}
```
Comment (billyvg) [inline tests/js/spec/views/organizationGroupDetails/groupTags.spec.jsx]: So we should still test that the API request gets called with the environment url params --
`MockApiClient.addMockResponse()` returns a jest mock function, which you can assert gets called with the right args.

**Review by billyvg — APPROVED** [review]

**Pull request merged**